README.md

```diff
@@ -1,2 +1,180 @@
+
+ENV['ember-cli-workbox'] = {
+  enabled: environment !== 'test',
+  debug
+};
+```
+* **enabled** - (Boolean) Addon is enabled. Default to true on production builds
+* **debug** - (Boolean) Log serviceworker states (registering, updating, etc)
+
+You can further customize ember-cli-workbox by setting **workbox configurations** in your environment.js file:
+
+```JavaScript
+//app/config/environment.js
+
+ENV['workbox'] = {
+  globPatterns: ['**\/*.{html,js,css}'],
+  globDirectory: './',
+  globIgnores: [],
+  ...
+};
+```
+
+* **swDest** - (String) The path to the final service worker file that will be created by the build process, relative to the build directory. Default path: './sw.js'
+* **globPatterns** - (Array of String) Files matching against any of these glob patterns will be included in the precache manifest. By default sw precaches all our ember application assets that match '**/*.{json,css,js,png,svg,eot,ttf,woff,jpg,gif,ico,xml,html,txt}'
+* **globDirectory** - (String) The base directory you wish to match globPatterns against, related to the build directory. Default  './'
+* **globIgnores** - (String or Array of String) Files matching against any of these glob patterns will be excluded from the file manifest, overriding any matches from globPatterns.
+E.g. globIgnores: ['**\/ignored.html']
+* **templatedUrls** - (Object with Array or String properties) If a URL is rendered generated based on some server-side logic, its contents may depend on multiple files or on some other unique string value.
+* **cacheId** - (String) An optional ID to be prepended to caches used by workbox-sw. This is primarily useful for local development where multiple sites may be served from the same http://localhost origin. Defaults to your app name (config.APP.name).
+* **runtimeCaching** - (Array of Object) Passing in an array of objects containing urlPatterns, handlers, and potentially options that will add the appropriate code to the generated service worker to handle runtime caching. The handler values correspond the names of the [strategies](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-sw.Strategies) supported by workbox-sw (cacheFirst, cacheOnly, networkFirst, networkOnly, staleWhileRevalidate)
+```JavaScript
+runtimeCaching: [{
+  // You can use a RegExp as the pattern:
+  urlPattern: /https://api.example.com/,
+  handler: 'cacheFirst',
+  // Any options provided will be used when
+  // creating the caching strategy.
+  options: {
+    cacheName: 'my-api-cache',
+    cacheExpiration: {
+      maxEntries: 10,
+    },
+  },
+},
+ ...
+]
+```
+* **maximumFileSizeToCacheInBytes** - (number) This value can be used to determine the maximum size of files that will be precached
+* **more** - For more details on Workbox configuration take a look at: [Workbox Google Developers](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-build)
+
+**Subscribe to events:**
```

Comment:
This section should have heading, it's important to know how to attach code to the worker events, I mean that is not configuration.
Add an example to show how to subscribe to an event (I dont't have to know that the service extends from `Ember.Evented`):

```markdown
### Subscribing to events

The Service Worker will emit the next events:

- ** The list of events**

**Example of the event**
```